airbyte-ci/connectors/regression-testing/src/regression_testing/comparators/diff_comparator.py

```diff
@@ -0,0 +1,37 @@
+import os
+
+from .base_comparator import BaseComparator
+from ..utils import ConnectorUnderTest, sh_dash_c
+from dagger import Client, Container
+
+
+class DiffComparator(BaseComparator):
+
+    def __init__(self, dagger_client: Client, test_output_directory: str):
+        self._test_output_directory = test_output_directory
+        self._in_container_results_directory = "/tmp"
+        self._in_container_results_file = f"{self._in_container_results_directory}/diff_output.txt"
+        self._container = (
+            dagger_client.container().from_("debian:buster")
+            .with_directory(self._in_container_results_directory,
+                            dagger_client.host().directory(self._test_output_directory)))
+        os.makedirs(self._in_container_results_directory, exist_ok=True)
+
+    async def _diff(self, container: Container, control_connector: ConnectorUnderTest, target_connector: ConnectorUnderTest) -> Container:
```

Comment:
If we say we are comparing raw outputs. Can we diff `stderr` and `stdout` of the control and the target?
let's say you can access the executed container with `control_connector.executed_container`.
Then you can mount the `stdout/stderr` as files to you container with `diff`:
```python
self._container
.with_new_file(f"{control_directory}/stdout.txt", await control_connector.executed_container.stdout())
.with_new_file(f"{target_directory}/stdout.txt", await target_connector.executed_container.stdout())
.with_new_file(f"{control_directory}/stderr.txt", await control_connector.executed_container.stderr())
.with_new_file(f"{target_directory}/stderr.txt", await target_connector.executed_container.stderr())
```

Reply:
It's interesting to realize that with this approach the connector output is never transiting on the host filesystem, which can be an interesting security point.

Reply:
As discussed I think it would make sense to put this behind a `debug` flag. Let me know when your changes to the connector runner are stable and I'll add it.